src/averageAges.js

```diff
@@ -55,7 +79,28 @@ function calculateWomenAverageAge(people, withChildren) {
  * @return {number}
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // write code here
+  const children = people.filter(person => {
+    const hasMother = people.some(potentialMother => (
+      potentialMother.name === person.mother
+    ));
+
+    if (!onlyWithSon || !hasMother) {
+      return hasMother;
+    }
+
+    const isSon = person.sex === 'm';
+
+    return isSon;
+  });
+
+  const avarageAgeDifference = children.reduce((ageDiffSum, child) => {
+    const childMother = people.find(mother => mother.name === child.mother);
+    const ageDiff = child.born - childMother.born;
+
+    return ageDiffSum + ageDiff;
+  }, 0) / children.length;
+
+  return avarageAgeDifference;
```

Comment:
You don't need this const here.

```suggestion
  return children.reduce((ageDiffSum, child) => {
    const childMother = people.find(mother => mother.name === child.mother);
    const ageDiff = child.born - childMother.born;

    return ageDiffSum + ageDiff;
  }, 0) / children.length;
```

Reply:
agree